### PR TITLE
✨ Data Mapper Consent Wizard Progress Endpoint (#296)

### PR DESCRIPTION
**Pull request by mistryrn**
## Description of Changes

Base feature branch for #296 , contains 3 PRs worth of work to implement the Data Mapper Consent Wizard Progress endpoint.

---
### PR 1 of 3: [🗃️ Add consentProgress to Data Model (#296) (PR 1)](https://github.com/OHCRN/platform/pull/721)

Adds `consentProgress` to the **Consent DAS Participant** model, to track the Participant's most recently completed Consent Wizard step, represented by the `ConsentCategory` of that step.

Also restores the `prisma-migrations.md` doc into the repo, a longstanding TODO item.

### Consent DAS
Adds `consentProgress` to the `Participant` model.
- Updates `schema.prisma` to add the new field
- Adds a migration to update the Participants table

### Docs
- Restores the `prisma-migrations.md` doc into the `/docs` directory of the repo, for easier access than visiting the Wiki

### Special Instructions
Before running these changes, you will need to run the new migration in `consent-das`:
```bash
# from consent-das
pnpm prisma:migrate:dev
```
---
### PR 2 of 3: [✨ Implement Data Mapper GET Consent Wizard Progress (#296) (PR 2)](https://github.com/OHCRN/platform/pull/722)

Implements the `GET /wizard/progress/:participantId` endpoint in Data Mapper, adds supporting types. Also fixes a small bug in `getConsentParticipantById` in Data Mapper.

### Data Mapper
* Adds `GET /wizard/progress/:participantId` router to Data Mapper
* Fixes small bug in Data Mapper `getConsentParticipantById` search service function (return parsed `ConsentParticipantResponse` instead of raw data which could have `null` instead of `undefined`)

### Types
* Add `GetConsentProgressResponse` response type to `types/dataMapper`

### Special Instructions
Before running these changes, you should rebuild the `types` package:
```bash
pnpm build:types
```
---
### PR 3 of 3: [✨ Update consentProgress Upon Completion of Consent Wizard Steps (#296) (PR 3)](https://github.com/OHCRN/platform/pull/734)

Updates Data Mapper `create.ts` service to make an additional call in each Consent Wizard step that updates the Consent Participant's `consentProgress` upon successful response submission.

Also fixes some small type errors related to new nullable fields on the ConsentParticipant model (consentProgress, assentFormAcknowledged).

### Data Mapper
- Make a call to `updateConsentParticipant` in each `create<>Consent` to set the consent progress to the appropriate value at each step of the wizard.

### Types
- Minor fixes to ConsentParticipant-related response types to support the newly added nullable field `consentProgress`, as well as a small fix for a missing nullable/transform related to `assentFormAcknowledged`.

### Special Instructions
Before running these changes, you should rebuild the `types` package:
```
pnpm build:types
```
---
## PR Readiness Checklist

- [x] "Expected Outcome(s)" in ticket have been met
- [x] Ticket number included in PR title
- [x] Connected ticket to PR
- [x] Labels added to PR for service name (`consent-api`, `data-mapper`, etc...), type (`chore`, `documentation`, etc...), status (`draft`, `on-hold`, etc...) **if applicable**
- [x] Manual testing completed
- [x] Builds locally without errors or warnings
- [ ] Tests are updated (if required) and passing
- [ ] PR feedback has been addressed **if applicable**
- [ ] New environment variables added to `.env.schema` files, `README.md`
- [x] Added copyrights to new files

### CODE REVIEW

File: apps/consent-das/prisma/schema.prisma [hunk 1]
```diff
@@ -43,6 +43,7 @@ model Participant {
   ParticipantResponse    ParticipantResponse[]
   Signature              Signature[]
   assentFormAcknowledged Boolean?
+  consentProgress        ConsentCategory?
```
Comment (mistryrn) [inline apps/consent-das/prisma/schema.prisma:46]: `types` package updates to support this new nullable field in Consent Participant response types are handled in PR 3 for this feature: https://github.com/OHCRN/platform/pull/734

File: docs/prisma-migrations.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
+# Prisma Schema Modifications and Migrations
```
Comment (mistryrn) [inline docs/prisma-migrations.md:1]: Restoring this doc into the `/docs` folder since we've been meaning to do that for a while 😛

Comment (anncatton) [reply docs/prisma-migrations.md:1]: thanks for restoring this 🙏

**Review by anncatton — APPROVED** [review]

**Review by anncatton — APPROVED** [review]
leggooooo 🚀

**Pull request merged**